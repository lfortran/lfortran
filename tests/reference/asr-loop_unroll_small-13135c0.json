{
    "basename": "asr-loop_unroll_small-13135c0",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/loop_unroll_small.f90",
    "infile_hash": "5a308fd86dceda8d2b6a3c4fcb2412bf4d1b3f043105d06956960db8",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-loop_unroll_small-13135c0.stdout",
    "stdout_hash": "cd2c0fc1bf903a6e7e8750c394ab1090855f4bc61ad6e9c2723794c1",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}