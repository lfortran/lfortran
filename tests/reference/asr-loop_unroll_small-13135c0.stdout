(TranslationUnit
    (SymbolTable
        1
        {
            loop_unroll_small:
                (Program
                    (SymbolTable
                        2
                        {
                            array:
                                (Variable
                                    2
                                    array
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Array
                                        (Integer 4)
                                        [((IntegerConstant 1 (Integer 4) Decimal)
                                        (IntegerConstant 4 (Integer 4) Decimal))]
                                        FixedSizeArray
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            first:
                                (Variable
                                    2
                                    first
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            i:
                                (Variable
                                    2
                                    i
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            last:
                                (Variable
                                    2
                                    last
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            print_subrout:
                                (Function
                                    (SymbolTable
                                        3
                                        {
                                            x:
                                                (Variable
                                                    3
                                                    x
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    print_subrout
                                    (FunctionType
                                        [(Integer 4)]
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 3 x)]
                                    [(Print
                                        [(StringFormat
                                            ()
                                            [(Var 3 x)]
                                            FormatFortran
                                            (Character -1 0 ())
                                            ()
                                        )]
                                        ()
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            x:
                                (Variable
                                    2
                                    x
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    loop_unroll_small
                    []
                    [(DoLoop
                        ()
                        ((Var 2 i)
                        (IntegerConstant 1 (Integer 4) Decimal)
                        (IntegerConstant 4 (Integer 4) Decimal)
                        ())
                        [(Assignment
                            (ArrayItem
                                (Var 2 array)
                                [(()
                                (Var 2 i)
                                ())]
                                (Integer 4)
                                ColMajor
                                ()
                            )
                            (Var 2 i)
                            ()
                        )]
                        []
                    )
                    (Assignment
                        (Var 2 first)
                        (IntegerConstant 1 (Integer 4) Decimal)
                        ()
                    )
                    (Assignment
                        (Var 2 last)
                        (IntegerConstant 4 (Integer 4) Decimal)
                        ()
                    )
                    (DoLoop
                        ()
                        ((Var 2 i)
                        (Var 2 first)
                        (Var 2 last)
                        ())
                        [(Assignment
                            (ArrayItem
                                (Var 2 array)
                                [(()
                                (Var 2 i)
                                ())]
                                (Integer 4)
                                ColMajor
                                ()
                            )
                            (IntegerBinOp
                                (ArrayItem
                                    (Var 2 array)
                                    [(()
                                    (Var 2 i)
                                    ())]
                                    (Integer 4)
                                    ColMajor
                                    ()
                                )
                                Add
                                (Var 2 i)
                                (Integer 4)
                                ()
                            )
                            ()
                        )]
                        []
                    )
                    (DoLoop
                        ()
                        ((Var 2 i)
                        (IntegerConstant 1 (Integer 4) Decimal)
                        (IntegerConstant 4 (Integer 4) Decimal)
                        ())
                        [(Assignment
                            (Var 2 x)
                            (ArrayItem
                                (Var 2 array)
                                [(()
                                (Var 2 i)
                                ())]
                                (Integer 4)
                                ColMajor
                                ()
                            )
                            ()
                        )
                        (SubroutineCall
                            2 print_subrout
                            ()
                            [((Var 2 x))]
                            ()
                        )]
                        []
                    )
                    (DoLoop
                        ()
                        ((Var 2 i)
                        (IntegerConstant 1 (Integer 4) Decimal)
                        (IntegerConstant 4 (Integer 4) Decimal)
                        ())
                        [(Assignment
                            (Var 2 x)
                            (ArrayItem
                                (Var 2 array)
                                [(()
                                (Var 2 i)
                                ())]
                                (Integer 4)
                                ColMajor
                                ()
                            )
                            ()
                        )
                        (If
                            (IntegerCompare
                                (Var 2 x)
                                NotEq
                                (IntegerBinOp
                                    (IntegerConstant 2 (Integer 4) Decimal)
                                    Mul
                                    (Var 2 i)
                                    (Integer 4)
                                    ()
                                )
                                (Logical 4)
                                ()
                            )
                            [(ErrorStop
                                ()
                            )]
                            []
                        )]
                        []
                    )]
                )
        })
    []
)
